exec/testing/fake_exec.go

```diff
@@ -45,6 +71,82 @@ func (fake *FakeExec) LookPath(file string) (string, error) {
 	return fake.LookPathFunc(file)
 }
 
+func getCaller(n int) string {
+	_, file, line, ok := runtime.Caller(n + 1)
+	if !ok {
+		return "???"
+	}
+	if i := strings.LastIndexAny(file, "/\\"); i >= 0 {
+		file = file[i+1:]
+	}
+	return fmt.Sprintf("%s:%d", file, line)
+}
+
+func (fake *FakeExec) fail(format string, args ...interface{}) {
+	msg := fmt.Sprintf(format, args...)
+	if fake.T == nil {
+		panic(msg)
+	}
+
+	// Don't re-fail from a "defer fexec.AssertExpectedCommands()" if we already failed once
+	if fake.T.Failed() {
+		return
+	}
+	fake.T.Fatal(msg)
+}
+
+// AddCommand adds a Cmd to be returned by a future fake.Command(...) call. Call
+// SetCombinedOutput or SetRunOutput on the result to specify its result.
+func (fake *FakeExec) AddCommand(cmd string, args ...string) *FakeCmd {
```

Comment:
If you want to add a command in a different way, you can just continue to use the existing FakeExec API. There is currently only 1 test program in all of kubernetes that wants to do something other than what the API above supports. (`pkg/volume/git_repo/git_repo_test.go`)